cuda_core/cuda/core/experimental/_program.py

```diff
@@ -24,11 +175,11 @@ class Program:
 
     """
 
-    __slots__ = ("_handle", "_backend", )
+    __slots__ = ("_handle", "_backend", "_options" )
     _supported_code_type = ("c++", )
     _supported_target_type = ("ptx", "cubin", "ltoir", )
 
-    def __init__(self, code, code_type):
+    def __init__(self, code, code_type, options : ProgramOptions = ProgramOptions()):
```

Comment:
This creates the `ProgramOptions` instance at method definition time.  Generally it's better to use a `options = None` pattern in the signature and then check for that in the body of the function.  That way, if an `options` is given, you don't need to create an object you'll never use.

Reply:
Yup we should do this, and use the `check_or_create_options` machinery to handle the default value `None` gracefully.